### ISSUE TITLE
Style Loss VGG features

### ISSUE THREAD

**Issue by preddy5** [tph9608/Point-Synthesis#1]
Hey @tph9608,
Could you please help me to figure out how style loss is being used as mentioned in the paper. Since, from what I understand from the code, while you extract vgg features, gram matrix loss is calculated in image space, not in feature space.

```
target_texture_img = target_p2i(target_exemplar).repeat(1, 3, 1, 1).to(device)
fstyle_loss = StyleLoss(target_texture_img)

```
```
input_density_img = input_soft_points.repeat(1, 3, 1, 1)
fstyle_loss(input_density_img)

```

Thank you,
Pradyumna.

**Comment by phtu-cs** [comment]
Sorry for the confusion in the code. The style loss is calculated both on image and feature space.

model, texture_losses, structure_losses,_, histogram_losses = get_style_model_and_losses(cnn, target_texture_img,texture_layers, structure_layers) The above line calculates style loss in feature space (here, it is called texture_loss).
texture_layers, structure_layers)


On Oct 10, 2019, at 8:40 AM, Pradyumna Reddy Chinthala <notifications@github.com<mailto:notifications@github.com>> wrote:


Hey @tph9608<https://github.com/tph9608>,
Could you please help me to figure out how style loss is being used as mentioned in the paper. Since, from what I understand from the code while you extract vgg features, style loss is calculated in image space not in feature space.

target_texture_img = target_p2i(target_exemplar).repeat(1, 3, 1, 1).to(device)
fstyle_loss = StyleLoss(target_texture_img)



input_density_img = input_soft_points.repeat(1, 3, 1, 1)
fstyle_loss(input_density_img)



Thank you,
Pradyumna.

**Comment by preddy5** [comment]
Thank you for the reply @tph9608 :)
